subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -562,7 +562,15 @@ public void setConfigurationContainer(ConfigurationContainer configurationContai
 
     @Override
     public Convention getConvention() {
-        // TODO (donat) deprecate after all internal usages have been eliminated
+        DeprecationLogger.deprecateMethod(Project.class, "getConvention()")
+            .replaceWith("Project.getExtensions()")
+            .willBeRemovedInGradle8()
+            .withUpgradeGuideSection(7, "project_getconvention_deprecation")
+            .nagUser();
+        return getInternalConvention();
+    }
+
+    public Convention getInternalConvention() {
```

Comment:
I would probably just explicitly call `extensibleDynamicObject.getConvention()` in both places vs adding another method to remove later.